flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlNodeConverters.java

```diff
@@ -46,19 +49,51 @@ public class SqlNodeConverters {
     @SuppressWarnings({"unchecked", "rawtypes"})
     public static Optional<Operation> convertSqlNode(
             SqlNode validatedSqlNode, ConvertContext context) {
-        SqlNodeConverter converter = CONVERTERS.get(validatedSqlNode.getClass());
-        if (converter != null) {
-            return Optional.of(converter.convertSqlNode(validatedSqlNode, context));
+        // match by class first
+        SqlNodeConverter classConverter = CLASS_CONVERTERS.get(validatedSqlNode.getClass());
```

Comment:
Have a question in here.
For the case that if we have two implemtation of `SqlNodeConverter`.
1: `SqlInsertConverter` which returns `Optional.of(SqlKind.INSERT)` in method `supportedSqlKinds`
2: `SqlRichInsertConverter`which implements `SqlNodeConverter<RichSqlInsert>` but haven't overriden the method `supportedSqlKinds` .
If both `SqlInsertConverter` and `SqlRichInsertConverter` are registered,  for  `RichSqlInsert`,
in here, it will then use  `SqlRichInsertConverter` to do conversion as it match by class first.
Is it expected?

Reply:
Yes, that's right. Class converter should be a higher priority than SqlKind converter.